libs/ui/layouts/src/lib/search-bar.tsx

```diff
@@ -46,100 +60,146 @@ function NoResults() {
 	);
 }
 
-function SearchResults({
-	searchQuery,
-	resetSearchQuery
-}: {
-	searchQuery: string;
-	resetSearchQuery: () => void;
-}) {
-	const session = useSession();
-	const user = session.data?.user;
+export function SearchBar() {
+	const isUser = useSession().status === "authenticated";
+	const [searchQuery, setSearchQuery] = useState("");
 	const search = { title: searchQuery, page: 1 };
-	const isUserAdmin = user?.role === "ADMIN";
+	const searchAuthor = { displayName: searchQuery, page: 1 };
```

Comment:
i stumbled here. There is too less differenciation between search and searchQuery and searchAuthor.

i suggest better variable names here like:
```suggestion
const [filterText, setFilterText] = useState("");

// For a generic search (e.g., content with titles)
const titleSearchParams = { title: filterText, page: 1 };

// For author-specific search
const authorSearchParams = { displayName: filterText, page: 1 };

```

or just directly search them in the trpc query: